kpops/manifests/argo.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+import enum
+
+from pydantic import BaseModel, Field
+
+try:
+    from typing import Any, Self  # pyright: ignore[reportAttributeAccessIssue]
+except ImportError:
+    from typing_extensions import Self
+
+
+def enrich_annotations(
+    helm_values: dict[str, Any], key: str, value: str
+) -> dict[str, Any]:
+    annotations = helm_values.setdefault("annotations", {})
+    annotations[key] = value
+    return helm_values
+
+
+class ArgoHook(str, enum.Enum):
+    POST_DELETE = "PostDelete"
+
+    @property
+    def key(self) -> str:
+        return "argocd.argoproj.io/hook"
+
+
+class ArgoSyncWave(BaseModel):
+    sync_wave: int = Field(default=0, serialization_alias="SyncWave")
```

Comment:
alias needed?
```suggestion
    sync_wave: int = 0
```

Reply:
is it possible to have negative sync waves? if not, I would be happy to further constrain this type, i.e. `pydantic.NonNegativeInt`

Reply:
They can be negative :)